waiter/src/waiter/state/maintainer.clj

```diff
@@ -689,8 +699,12 @@
                                              (select-keys new-state relevant-keys))
                                    ; propagate along router-state-push-chan only when state changes
                                    (async/put! router-state-push-chan new-state))
+                                 ; handle sending events asynchronously
+                                 (when (not (empty? all-new-failed-instances'))
+                                   (log/info "propagating failed instances events to internal handler" {:new-failed-instances-count (count all-new-failed-instances')})
+                                   (async/put! event-chan {:type :instance-failure :object all-new-failed-instances'}))
```

Comment:
Instead of pushing this out, I think we should leverage the `router-state-push-mult` and create a new go-block (what you currently call `start-event-maintainer`) that listens on state from here and does the failed instance diff calculation. The autoscaler is another example of a reader of this state.